src/rp2_common/pico_bootrom/include/pico/bootrom.h

```diff
@@ -1071,6 +1071,29 @@ static inline int rom_get_last_boot_type(void) {
  */
 int rom_add_flash_runtime_partition(uint32_t start_offset, uint32_t size, uint32_t permissions);
 
+/*! \brief  Pick A/B partition with TBYB guards
```

Comment:
I still find the function description a bit confusing:
* maybe something along the lines of "Pick A/B partition for an arbitrary pair of partitions without disturbing any in process update A/B or TBYB boot" (needs improvement still)...
* not sure people know what a flash_erase_addr is maybe describe it in words (e.g. the erase of xxx because yyy"
* I'd say "using the flash_update_boot_window_base from the current boot"

Reply:
I think "Pick A/B partition without disturbing any in progress update or TBYB boot" works for the description - pick_ab_partition is always for an arbitrary pair of partitions, so that probably doesn't need including in the description?

Reply:
Ah, I was (wrongly) thinking this new API existed because pick_ab_partition was for the regular boot partitions, rather than because that API is somewhat broken. Perhaps just add a warning to the other API referencing this one